lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -22,11 +33,80 @@ public MediaItemsController(Library library) throws IOException {
     this.librarian = library.getLibrarians().stream().findFirst().orElseThrow();
   }
 
-  @GetMapping("/items")
-  public GetMediaItemsResponse getItems() {
+  @GetMapping()
+  public ResponseEntity<GetMediaItemsResponse> getItemById() {
     Set<MediaItem> items = library.search(SearchCriteria.builder().build());
+    if (items.isEmpty()) {
+      ResponseEntity.noContent();
+    }
     List<MediaItemResponse> responseItems = items.stream().map(MediaItemResponse::from).toList();
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
-    return response;
+    return ResponseEntity.ok(response);
+  }
+
+  @GetMapping(value = "/{id}")
+  public ResponseEntity<MediaItemResponse> getItemById(@PathVariable("id") UUID id) {
+
+    System.out.println(id.toString());
+
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id.toString()).build());
+    Optional<MediaItem> matchedItem =
+        items.stream().filter(item -> item.getId().equals(id)).findFirst();
+    System.out.println("items");
+    System.out.println(items);
+    System.out.println(matchedItem);
+    System.out.println("mathcedItems");
+    return matchedItem
+        .map(
+            item -> {
+              MediaItemResponse responseItem = MediaItemResponse.from(item);
+              return ResponseEntity.ok(responseItem);
+            })
+        .orElse(ResponseEntity.notFound().build());
+  }
+
+  @PostMapping()
+  public ResponseEntity<?> postItems(@RequestBody CreateMediaItemRequest requestBody) {
+    List<String> errorsList = new ArrayList<>();
+    MediaItemRequest request = requestBody.getItem();
+    if (requestBody.getItem() == null) {
```

Comment:
Won't need any of this validation code if you just use the `@Valid` attribute on the `requestBody` argument.